services/rpcfilters/latest_logs.go

```diff
@@ -0,0 +1,89 @@
+package rpcfilters
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// ContextCaller provides CallContext method as ethereums rpc.Client.
+type ContextCaller interface {
+	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
+}
+
+func pollLogs(client ContextCaller, f *logsFilter, timeout, period time.Duration) {
```

Comment:
What do you think of replacing `timeout` with `context.Context`? This way it will be possible to cancel `pollLogs` by the caller.

Reply:
but timeout used for every new query. maybe i misunderstood you

Reply:
You're right, I did not notice that.